### DISCUSSION TITLE
How to align images on left side using page.insert_image

### DISCUSSION THREAD

**Discussion by MallRoy** [pymupdf/PyMuPDF#2149] [Looking for help] [answered]
I am using ```page.insert_image(rect, stream=stre, xref=0,rotate=page.rotation)```  for inserting images. When I place the images, the images are center aligned. is there a way to specify the image alignment?

**Comment by JorjMcKie** [comment, marked as answer]
> I am using `page.insert_image(rect, stream=stre, xref=0,rotate=page.rotation)` for inserting images. When I place the images, the images are center aligned. is there a way to specify the image alignment?

No, not directly. But indirectly, you have ways to achieve what you want:
The method always exploits at least one of width or height of your `rect`. As you indicated, you are using memory-based images. This makes it easy to determine the image's width and height, e.g. by using `fitz.image_profile`:
```python
import fitz
import pathlib
stream = pathlib.Path("nur-ruhig.jpg").read_bytes()
from pprint import pprint
pprint(fitz.image_profile(stream))
{'bpc': 8,
 'colorspace': 3,
 'cs-name': 'DeviceRGB',
 'ext': 'jpeg',
 'height': 501,
 'orientation': 0,
 'transform': (1.0, 0.0, 0.0, 1.0, 0.0, 0.0),
 'width': 439,
 'xres': 96,
 'yres': 96}
```
If you want to make sure, that the image is aligned at `rect.x0`, consider that `rect` and image should have the same aspect ratio. Therefore make sure that `height / width = (rect.y1 -rect.y0) / rect.width`.
This leads to setting `rect.y1 = height / width * rect.width + rect.y0`.

That will cause the image occupying (original) rect with its top-left corner at `rect.tl`.
Similarly trivial calculations let you achieve bottom-left alignment, or even left alignment vertically centered.
